spec/components/email_alert_fieldset_component_spec.rb

```diff
@@ -15,11 +15,69 @@
     expect(page).to have_checked_field("email_alerts", with: "no")
   end
 
+  it "sets the value of the radio button to 'no' if the schema does not have a signup_content_id" do
+    schema = FinderSchema.new
+    render_inline(described_class.new(schema:))
+
+    expect(page).to have_checked_field("email_alerts", with: "no")
+  end
+
   it "sets the value of the radio button to 'all_content' if the schema has a signup content id" do
     schema = FinderSchema.new
     schema.signup_content_id = "123"
     render_inline(described_class.new(schema:))
 
     expect(page).to have_checked_field("email_alerts", with: "all_content")
+    expect(page).to have_field("signup_content_id", with: "123", type: "hidden")
+  end
+
+  it "sets the value of the radio button to 'all_content' and renders a hidden input with a generated content id if the schema is missing a signup content id" do
+    schema = FinderSchema.new
+    allow(SecureRandom).to receive(:uuid).and_return("new-id")
+    render_inline(described_class.new(schema:))
+
+    expect(page).to have_checked_field("email_alerts", with: "no")
+    expect(page).to have_field("signup_content_id", with: "new-id", type: "hidden")
+  end
+
+  it "renders the email subscription topic if the all_content value is checked" do
+    schema = FinderSchema.new
+    schema.signup_content_id = "123"
+    schema.subscription_list_title_prefix = "Finder email subscription"
+    render_inline(described_class.new(schema:))
+
+    expect(page).to have_text("Email subscription topic")
+    expect(page).to have_field("subscription_list_title_prefix", with: schema.subscription_list_title_prefix)
+  end
+
+  it "does not render the email subscription topic if the all_content value is checked but the current schema value is a hash" do
```

Comment:
Worth adding a similar comment to what you added earlier, to imply the test is temporary and can be deleted one day?